dipy/reconst/dki.py

```diff
@@ -700,8 +800,427 @@ def split_dki_param(dki_params):
+                   6*(alpha+1)*U[0, 0]*V22 - 6*(alpha+1)*U[2, 2]*V00 - \
+                   24*(alpha+1)*U[0,2]*V02 + 2*(alpha+1)*(alpha+3)*V00*V22 + \
+                   4*(alpha+1)*(alpha+3)*V02*V02) + \
+         kt[11] * (6*U[1, 1]*U[2, 2] + 12*U[1, 2]*U[1, 2] - \
+                   6*(alpha+1)*U[1,1]*V22 - 6*(alpha+1)*U[2, 2]*V11 - \
+                   24*(alpha+1)*U[1,2]*V12 + 2*(alpha+1)*(alpha+3)*V11*V22 + \
+                   4*(alpha+1)*(alpha+3)*V12*V12) + \
+         kt[12] * (12*U[0, 0]*U[1, 2] + 24*U[0, 1]*U[0, 2] - \
+                   12*(alpha+1)*U[0, 0]*V12 - 12*(alpha+1)*U[1, 2]*V00 - \
+                   24*(alpha+1)*U[0, 1]*V02 - 24*(alpha+1)*U[0, 2]*V01 + \
+                   4*(alpha+1)*(alpha+3)*V00*V12 + \
+                   8*(alpha+1)*(alpha+3)*V01*V02) + \
+         kt[13] * (12*U[1, 1]*U[0, 2] + 24*U[1, 0]*U[1, 2] - \
+                   12*(alpha+1)*U[1, 1]*V02 - 12*(alpha+1)*U[0, 2]*V11 - \
+                   24*(alpha+1)*U[0, 1]*V12 - 24*(alpha+1)*U[1, 2]*V01 + \
+                   4*(alpha+1)*(alpha+3)*V11*V02 + \
+                   8*(alpha+1)*(alpha+3)*V01*V12) + \
+         kt[14] * (12*U[2, 2]*U[0, 1] + 24*U[2, 0]*U[2, 1] - \
+                   12*(alpha+1)*U[2, 2]*V01 - 12*(alpha+1)*U[0, 1]*V22 - \
+                   24*(alpha+1)*U[0, 2]*V12 - 24*(alpha+1)*U[1, 2]*V02 + \
+                   4*(alpha+1)*(alpha+3)*V22*V01 + \
+                   8*(alpha+1)*(alpha+3)*V02*V12)))
+
+    return ODF
+
+
 common_fit_methods = {'WLS': wls_fit_dki,
                       'OLS': ols_fit_dki,
                       'UWLLS': wls_fit_dki,
                       'ULLS': ols_fit_dki,
                       }
+
+ind_ele = {1: 0, 16: 1, 81: 2, 2: 3, 3: 4, 8: 5, 24: 6, 27: 7, 54: 8, 4: 9,
+           9: 10, 36: 11, 6: 12, 12: 13, 18: 14}
+
+
+def Wcons(k_elements):
+    r""" Construct the full 4D kurtosis tensors from its 15 independent
+    elements
+    
+    Parameters
+    ----------
+    k_elements : (15,)
+        elements of the kurtosis tensor in the following order:
+        
+        .. math::
+            
+    \begin{matrix} ( & W_{xxxx} & W_{yyyy} & W_{zzzz} & W_{xxxy} & W_{xxxz}
+                     & ... \\
+                     & W_{xyyy} & W_{yyyz} & W_{xzzz} & W_{yzzz} & W_{xxyy}
+                     & ... \\
+                     & W_{xxzz} & W_{yyzz} & W_{xxyz} & W_{xyyz} & W_{xyzz}
+                     & & )\end{matrix}
+    Returns
+    -------
+    W : array(3, 3, 3, 3)
+        Full 4D kurtosis tensor
+    """
+
+    W = np.zeros((3, 3, 3, 3))
+
+    xyz = [0, 1, 2]
+    for ind_i in xyz:
+        for ind_j in xyz:
+            for ind_k in xyz:
+                for ind_l in xyz:
+                    key = (ind_i+1) * (ind_j+1) * (ind_k+1) * (ind_l+1)
+                    W[ind_i][ind_j][ind_k][ind_l] = k_elements[ind_ele[key]]
+
+    return W
+
+
+def dki_directions(dki_params, sphere, alpha=4, relative_peak_threshold=0.1,
```

Comment:
We need to think about how to make this play nicely with this function: 

https://github.com/nipy/dipy/blob/master/dipy/reconst/peaks.py#L325